### PR TITLE
Adds an autocannon ship gun

### PR DESCRIPTION
**Pull request by hazelrat**
This adds the Goshawk heavy autocannon to the game, intended as a generic gun emplacement available to be mapped into any appropriate offship.

The AP shots are intended to be the weakest, but they do a lot of damage on the off-chance they do hit someone. AP fires twelve shots, frag fires nine, and HE only fires six - obviously ICly every salvo is the same size, this is just for balance. As is currently, I'd say this is a little stronger and generally a bit more versatile than the grauwolf.

**This is up for review.**

Credit to Noble for the spriting!

### CODE REVIEW

**Comment by hazelrat** [comment]
!wip

**Comment by hazelrat** [comment]
!review

**Review by Geevies — APPROVED** [review]

File: code/modules/overmap/ship_weaponry/weaponry/autocannon.dm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	icon_state = "small"
+	damage = 60
+	armor_penetration = 30
+	penetrating = 2
+
+/obj/projectile/ship_ammo/autocannon/he/on_hit(atom/target, blocked, def_zone, is_landmark_hit)
+	. = ..()
+	explosion(target, 0, 2, 3)
+
+/obj/projectile/ship_ammo/autocannon/frag/on_impact(var/atom/A)
```
Comment (FluffyGhoster) [inline code/modules/overmap/ship_weaponry/weaponry/autocannon.dm:82]: This proc doesn't exist anymore and needs to be changed to either `Impact()` or consolidated with `on_hit()`

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

File: code/modules/overmap/ship_weaponry/weaponry/autocannon.dm (hunk 1 above)
Comment (hazelrat) [reply code/modules/overmap/ship_weaponry/weaponry/autocannon.dm]: Refactored to use on_hit, works in testing.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**